computer_vision/mosaic_augmentation.py

```diff
@@ -0,0 +1,180 @@
+    scale_y = scale_range[0] + \
+        random.random() * (scale_range[1] - scale_range[0])
+    divid_point_x = int(scale_x * output_size[1])
+    divid_point_y = int(scale_y * output_size[0])
+
+    new_anno = []
+    path_list = []
+    for i, idx in enumerate(idxs):
+        path = all_img_list[idx]
+        path_list.append(path)
+        img_annos = all_annos[idx]
+        img = cv2.imread(path)
+        if i == 0:  # top-left
+            img = cv2.resize(img, (divid_point_x, divid_point_y))
+            output_img[:divid_point_y, :divid_point_x, :] = img
+            for bbox in img_annos:
+                xmin = bbox[1] * scale_x
+                ymin = bbox[2] * scale_y
+                xmax = bbox[3] * scale_x
+                ymax = bbox[4] * scale_y
+                new_anno.append([bbox[0], xmin, ymin, xmax, ymax])
+        elif i == 1:  # top-right
+            img = cv2.resize(
+                img, (output_size[1] - divid_point_x, divid_point_y))
+            output_img[:divid_point_y, divid_point_x:output_size[1], :] = img
+            for bbox in img_annos:
+                xmin = scale_x + bbox[1] * (1 - scale_x)
+                ymin = bbox[2] * scale_y
+                xmax = scale_x + bbox[3] * (1 - scale_x)
+                ymax = bbox[4] * scale_y
+                new_anno.append([bbox[0], xmin, ymin, xmax, ymax])
+        elif i == 2:  # bottom-left
+            img = cv2.resize(
+                img, (divid_point_x, output_size[0] - divid_point_y))
+            output_img[divid_point_y:output_size[0], :divid_point_x, :] = img
+            for bbox in img_annos:
+                xmin = bbox[1] * scale_x
+                ymin = scale_y + bbox[2] * (1 - scale_y)
+                xmax = bbox[3] * scale_x
+                ymax = scale_y + bbox[4] * (1 - scale_y)
+                new_anno.append([bbox[0], xmin, ymin, xmax, ymax])
+        else:  # bottom-right
+            img = cv2.resize(
+                img, (output_size[1] - divid_point_x, output_size[0] - divid_point_y))
+            output_img[divid_point_y:output_size[0],
+                       divid_point_x:output_size[1], :] = img
+            for bbox in img_annos:
+                xmin = scale_x + bbox[1] * (1 - scale_x)
+                ymin = scale_y + bbox[2] * (1 - scale_y)
+                xmax = scale_x + bbox[3] * (1 - scale_x)
+                ymax = scale_y + bbox[4] * (1 - scale_y)
+                new_anno.append([bbox[0], xmin, ymin, xmax, ymax])
+
+    # Remove bounding box small than scale of filter
+    if 0 < filter_scale:
+        new_anno = [anno for anno in new_anno if
+                    filter_scale < (anno[3] - anno[1]) and filter_scale < (anno[4] - anno[2])]
+
+    return output_img, new_anno, path_list[0]
+
+
+def random_chars(number_char: int) -> str:
+    """
+    Automatic generate random 32 characters.
+    Get random string code: '7b7ad245cdff75241935e4dd860f3bad'
+    >>> random_chars(32)
+    '7b7ad245cdff75241935e4dd860f3bad'
```

Comment:
On your machine, try to run `python3 -m doctest -v computer_vision/mosaic_augmentation.py` to see if the tests pass.